apps/condo/domains/user/schema/ForgotPasswordService.js

```diff
@@ -133,25 +135,49 @@ const ForgotPasswordService = new GQLCustomSchema('ForgotPasswordService', {
                 if (password.length < MIN_PASSWORD_LENGTH) {
                     throw new Error(`${PASSWORD_TOO_SHORT}] Password is too short`)
                 }
-                const [action] = await ForgotPasswordActionUtil.getAll(context, {
+
+                let [action] = await ForgotPasswordActionUtil.getAll(context, {
                     token,
                     expiresAt_gte: now,
                     usedAt: null,
+                }) || await ConfirmPhoneActionUtil.getAll(context, {
+                    token, 
+                    expiresAt_gte: now,
+                    completedAt: null,
                 })
 
                 if (!action) {
                     throw new Error(`${RESET_TOKEN_NOT_FOUND}] Unable to find valid token`)
                 }
 
-                const userId = action.user.id
-                const { phone } = await getById('User', userId)
-                const tokenId = action.id
+                let phone, userId
 
-                // mark token as used
-                await ForgotPasswordActionUtil.update(context, tokenId, {
-                    usedAt: now,
-                })
+                if (typeof action.smsCode === 'number') {
+                    if (action.isPhoneVerified === false) {
+                        throw new Error(`${RESET_TOKEN_NOT_FOUND}] Unable to find valid token`)
+                    }
+
+                    const userId = await User.getAll(context, {
+                        phone,
+                    }).then(p => p.id)
 
+                    if (!userId) {
+                        throw new Error(`${RESET_TOKEN_NOT_FOUND}] Unable to find valid token`)
+                    }
+
+                    phone = action.phone
+                }
+                else if (action.user) {
+                    const userId = action.user.id
+                    phone = await getById('User', userId).then(p => p.phone)
+                    const tokenId = action.id
+
+                    // mark token as used
+                    await ForgotPasswordActionUtil.update(context, tokenId, {
+                        usedAt: now,
+                    })
+                }
+                
```

Comment:
I think we need to set `ConfirmPhoneAction.completedAt` here to current date. But this flag is not set anywhere at all :/